lib/restAPIs/datasets.js

```diff
@@ -11,13 +12,11 @@ module.exports = class RestTopic extends RestUtil {
     return 'datasets';
   }
 
-  getDatasetDefinition(datasetId, fields = []) {
+  getDatasetDefinition(datasetId, fields) {
     assert(datasetId, 'No datasetId provided');
 
-    const joinedFields = fields.join('&fields=');
-    const queryString = joinedFields ? 'fields=' + joinedFields : false;
-
-    return this._makeRequest(`${this._getBase()}/${datasetId}`, queryString, 'GET', this.getProjectHeader())
+    const queryParameter = buildCommonQuery(false, false, fields);
```

Comment:
Great idea! Since fields is used a lot more than the other two, it may be worth having "fields" be the first parameter and the other two are after that. When you only want fields, you could then do:
```
    const queryParameter = buildCommonQuery(fields);
```

If you did want all 3, then it's just a matter of adding more params:
```
    const queryParameter = buildCommonQuery(fields, a, b);
```